### DISCUSSION TITLE
Potential change of default settings: when in insert mode the cursor should be a line

### DISCUSSION THREAD

**Discussion by godalming123** [helix-editor/helix#8634] [Polls]
I think that this should be the default cursor shape settings:
```
[editor.cursor-shape]
insert = "bar"
normal = "block"
select = "underline"
```
What do you think?
*Note:* I have voted for yes on this discussion

**Comment by pascalkuthe** [comment]
I don't think the default cursor shape should ever be anything but a block cursor.

Terminals are limited in that they can only have a single primary cursor. That means we always need to render the other cursors as block cursors. As such we will not change the actual semantics of the cursor. The cursor is *always* a block the cursor shape is *only visual*.

As such showing anything but a block cursor by default is misleading. There were multiple people reporting bugs which were simply explained by a misunderstanding of our cursor semantics because they used this setting. So it should definitely not be the default unless we switch to zero-width cursors (which is blocked on having a GUI IMO).
